### DISCUSSION TITLE
Jpeg autorotate

### DISCUSSION THREAD

**Discussion by zhifengzhuang** [libvips/libvips#3268] [Q&A] [answered]
Hello @jcupitt .I have a jpeg with orientation. After I use vips_jpegload and set autorotate true, I got unexpected result.
The original image is: 
![xx](https://user-images.githubusercontent.com/13492988/211235807-31517392-7819-400f-a807-ebea31afe887.jpeg)
After I  run cmd: vips jpegload xx.jpeg out.jpg --autorotate=true, I got the result:
![out](https://user-images.githubusercontent.com/13492988/211235893-157ed7b9-ffd3-4377-b68f-e01e85dd91d1.jpg)

However，the cmd bellow seems to generate a expected result: vips autorot xx.jpeg autorot.jpeg
![autorot](https://user-images.githubusercontent.com/13492988/211236006-0353f97f-74ee-43ef-be09-178b3b45b388.jpeg)

My vips version is: vips-8.13.3

**Comment by jcupitt** [comment, marked as answer]
Hi @zhifengzhuang,

You're right, there was a logic error in the autorotate code if width and height didn't change with orientation. I've fixed this for 8.14.1.

Thanks for reporting this!

**Comment by zhifengzhuang** [comment]
Thanks for answering @jcupitt ,
another question is what is the best way to auto orientation for a jpeg?Vips jpegload autorotate true or vips autorot?Any difference between them?

**Reply by jcupitt** [reply]
If you are downsizing images, it's faster to do the rotate after the shrink, since the image is smaller then.

If the image size isn't changing, then it probably doesn't matter where you rotate. You'd need to do some tests.
